libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/idp/SPConfig.java

```diff
@@ -0,0 +1,171 @@
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission of salesforce.com, inc.
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.salesforce.androidsdk.auth.idp;
+
+import android.os.Bundle;
+
+import com.salesforce.androidsdk.auth.OAuth2;
+
+/**
+ * This class performs represents an SP app's config.
+ *
+ * @author bhariharan
+ */
+public class SPConfig {
+
+    private static final String OAUTH_CLIENT_ID_KEY = "oauth_client_id";
+    private static final String OAUTH_CALLBACK_URL_KEY = "oauth_callback_url";
+    private static final String CODE_CHALLENGE_KEY = "code_challenge";
+    private static final String OAUTH_SCOPES_KEY = "oauth_scopes";
+    private static final String LOGIN_URL_KEY = "login_url";
+    private static final String USER_HINT_KEY = "user_hint";
+    private static final String COMPUTED_SCOPE_PARAMETER_KEY = "computed_scope_parameter";
+
+    private String oauthClientId;
+    private String oauthCallbackUrl;
+    private String codeChallenge;
+    private String[] oauthScopes;
+    private String loginUrl;
+    private String userHint;
+    private String computedScopeParameter;
+
+    /**
+     * Parameterized constructor.
+     *
+     * @param oauthClientId OAuth client ID.
+     * @param oauthCallbackUrl OAuth callback URL.
+     * @param codeChallenge Code challenge.
+     * @param oauthScopes OAuth scopes.
+     * @param loginUrl Login URL.
+     * @param userHint User hint.
+     */
+    public SPConfig(String oauthClientId, String oauthCallbackUrl, String codeChallenge,
+                    String[] oauthScopes, String loginUrl, String userHint) {
+        this.oauthClientId = oauthClientId;
+        this.oauthCallbackUrl = oauthCallbackUrl;
+        this.codeChallenge = codeChallenge;
+        this.oauthScopes = oauthScopes;
+        this.loginUrl = loginUrl;
+        this.userHint = userHint;
+        computedScopeParameter = OAuth2.computeScopeParameter(oauthScopes);
+    }
+
+    /**
+     * Parameterized constructor for internal use only.
+     *
+     * @param config Bundle containing config.
+     */
+    SPConfig(Bundle config) {
```

Comment:
Convenience methods to make and extract from a `Bundle` to easily pass this object between apps and/or activities.